gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,341 @@
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.VersionDoesNotExistException;
+import gobblin.util.PathUtils;
+
+
+/**
+ * An implementation of {@link ConfigStore} backed by HDFS. The class assumes a simple file and directory layout
+ * structure where each path under the root store directory corresponds to a dataset. The {@link #getStoreURI()} method
+ * gives an {@link URI} that identifies the HDFS cluster being used, as well as the root directory of the store. When
+ * querying this store, the scheme should be of the form {@code simple-[hdfs-scheme]} (a.k.a the logical scheme). For
+ * example, if the store is located on a the local filesystem the scheme should be {@code simple-file}, if the store
+ * is located on HDFS, the scheme should be {@code simple-hdfs}. This class can be constructed using a
+ * {@link SimpleHDFSConfigStoreFactory}.
+ *
+ * <p>
+ *   The class assumes a directory called {@link #CONFIG_STORE_NAME} is under the root directory. This folder should
+ *   contain a directory for each version deployed to the {@link ConfigStore}. An example directory structure could look
+ *   like: <br>
+ *   <blockquote>
+ *     <code>
+ *       /root<br>
+ *       &emsp;/my-simple-store<br>
+ *       &emsp;&emsp;/_CONFIG_STORE<br>
+ *       &emsp;&emsp;&emsp;/v1.0<br>
+ *       &emsp;&emsp;&emsp;&emsp;/dataset1<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/child-dataset<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/includes.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;/dataset2<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;/child-dataset<br>
+ *       &emsp;&emsp;&emsp;&emsp;&emsp;&emsp;/main.conf<br>
+ *     </code>
+ *   </blockquote>
+ * </p>
+ *
+ * <p>
+ *   In the above example, the root of the store is {@code /root/my-simple-store/}. The code automatically assumes that
+ *   this folder contains a directory named {@link #CONFIG_STORE_NAME}. In order to access the dataset
+ *   {@code dataset1/child-dataset} using {@link ConfigClient#getConfig(URI)}, the specified {@link URI} should be
+ *   {@code simple-hdfs://[authority]:[port]/root/my-simple-store/dataset1/child-dataset/}. Note this is the fully
+ *   qualified path to the actual {@code main.conf} file on HDFS, with the {@link #CONFIG_STORE_NAME} and the
+ *   {@code version} directories removed.
+ * </p>
+ *
+ * <p>
+ *   All the {@link Config}s for a dataset should be put in the associated {@code main.conf} file, and all the imports
+ *   should be put in the associated {@code includes.conf} file.
+ * </p>
+ *
+ * <p>
+ *   This class is not responsible for deploying configurations from an external source to HDFS, only for reading them.
+ * </p>
+ *
+ * @see {@link SimpleHDFSConfigStoreFactory}
+ */
+public class SimpleHDFSConfigStore implements ConfigStore {
+
+  public static final String SIMPLE_HDFS_SCHEME_PREFIX = "simple-";
```

Comment:
It's basically the prefix of the logical store. For `SimpleHDFSConfigStore` the logical scheme is `simple-hdfs` for `SimpleLocalHDFSConfigStore` the logical scheme is `simple-local`. The name `simple-` is arbitrary. I thought it useful to have a prefix since there may be multiple `ConfigStore`s that right to HDFS. We may want to create additional HDFS backed `ConfigStore`s that are more efficient.